### PR TITLE
Added git.remote_branch, git.behind, git.is_behind

### PR DESCRIPTION
**Pull request by Lugoues**
Just a few helpers i found useful when checking to see if i needed to git pull and run other update code.

### CODE REVIEW

**Comment by groggemans** [comment]
Hi Peter,

Thank you for the PR!

Would you mind to also add these to the API docs? (`docs/api.md`)

File: src/git.bash [hunk 1]
```diff
@@ -42,6 +47,18 @@ git.ahead() {
     git status -sb --porcelain | grep -o '\[.*\]'
 }
 
+# Print how far behind gir repo is
```
Comment (groggemans) [inline src/git.bash]: That should probably be `git` instead of `gir` :)

**Comment by Lugoues** [comment]
done! Fixed a few other issues I saw in the api doc as well :)

**Comment by groggemans** [comment]
Thank you!

**Pull request merged**